libethereum/ExtVM.cpp

```diff
@@ -95,30 +94,62 @@ void go(unsigned _depth, Executive& _e, OnOpFunc const& _onOp)
 
 bool ExtVM::call(CallParameters& _p)
 {
-	Executive e(m_s, envInfo(), m_sealEngine, depth + 1);
+	Executive e{m_s, envInfo(), m_sealEngine, depth + 1};
 	if (!e.call(_p, gasPrice, origin))
 	{
 		go(depth, e, _p.onOp);
 		e.accrueSubState(sub);
 	}
 	_p.gas = e.gas();
 
-	return !e.excepted();
+	if (e.excepted())
+		return false;
+
+	// FIXME: Make sure move constructor is really used.
+	m_orig.children.emplace_back(e.takeSnapshot());
+	return true;
 }
 
 size_t ExtVM::codeSizeAt(dev::Address _a)
 {
 	return m_s.codeSize(_a);
 }
 
+void ExtVM::setStore(u256 _n, u256 _v)
```

Comment:
This is just moved implementation from the header file. I will not change the name in this PR.